miner/multi_worker.go

```diff
@@ -73,14 +76,35 @@ func (w *multiWorker) disablePreseal() {
 	}
 }
 
-func newMultiWorker(config *Config, chainConfig *params.ChainConfig, engine consensus.Engine, eth Backend, mux *event.TypeMux, isLocalBlock func(*types.Block) bool, init bool) *multiWorker {
+func newMultiWorker(config *Config, chainConfig *params.ChainConfig, engine consensus.Engine, eth Backend, mux *event.TypeMux, isLocalBlock func(*types.Block) bool, init bool, jsonMEVLogFile *os.File) *multiWorker {
 	queue := make(chan *task)
 
 	regularWorker := newWorker(config, chainConfig, engine, eth, mux, isLocalBlock, init, &flashbotsData{
 		isFlashbots: false,
 		queue:       queue,
 	})
 
+	if jsonMEVLogFile != nil {
+		type r struct {
+			blockNumber *big.Int
+			createdAt   time.Time
+			profit      *big.Int
+			isFlashbots bool
+			mevTxs      types.Transactions
+		}
+
+		regularWorker.newTaskHook = func(t *task) {
+			if err := json.NewEncoder(jsonMEVLogFile).Encode(r{
```

Comment:
I think we should also include the worker number, which tells us how many bundles were merged.

Reply:
lol - I did have it initially but then it was more plumbing - true though, I'll put it in